test/common/quic/platform/quic_platform_test.cc

```diff
@@ -474,33 +475,34 @@ TEST_F(QuicPlatformTest, MonotonicityWithFakeEpollClock) {
 }
 
 TEST_F(QuicPlatformTest, QuicFlags) {
-  auto& flag_registry = quiche::FlagRegistry::getInstance();
+  quiche::FlagRegistry::getInstance();
   EXPECT_TRUE(GetQuicReloadableFlag(quic_default_to_bbr));
-  flag_registry.resetFlags();
-
-  EXPECT_FALSE(GetQuicReloadableFlag(quic_testonly_default_false));
-  EXPECT_TRUE(GetQuicReloadableFlag(quic_testonly_default_true));
-  SetQuicReloadableFlag(quic_testonly_default_false, true);
-  EXPECT_TRUE(GetQuicReloadableFlag(quic_testonly_default_false));
 
-  EXPECT_FALSE(GetQuicRestartFlag(quic_testonly_default_false));
-  EXPECT_TRUE(GetQuicRestartFlag(quic_testonly_default_true));
-  SetQuicRestartFlag(quic_testonly_default_false, true);
-  EXPECT_TRUE(GetQuicRestartFlag(quic_testonly_default_false));
-
-  EXPECT_EQ(200, GetQuicFlag(FLAGS_quic_time_wait_list_seconds));
-  SetQuicFlag(FLAGS_quic_time_wait_list_seconds, 100);
-  EXPECT_EQ(100, GetQuicFlag(FLAGS_quic_time_wait_list_seconds));
+  {
+    quiche::test::QuicheFlagSaver saver;
+    EXPECT_FALSE(GetQuicReloadableFlag(quic_testonly_default_false));
+    EXPECT_TRUE(GetQuicReloadableFlag(quic_testonly_default_true));
+    SetQuicReloadableFlag(quic_testonly_default_false, true);
+    EXPECT_TRUE(GetQuicReloadableFlag(quic_testonly_default_false));
+
+    EXPECT_FALSE(GetQuicRestartFlag(quic_testonly_default_false));
+    EXPECT_TRUE(GetQuicRestartFlag(quic_testonly_default_true));
+    SetQuicRestartFlag(quic_testonly_default_false, true);
+    EXPECT_TRUE(GetQuicRestartFlag(quic_testonly_default_false));
```

Comment:
seems like this code is just a copy of above, from surrounding context I would have assumed this would have set the `quic_testonly_default_true` to false s.t. it gets restored to true when we leave the scope of flag saver.

Reply:
Good catch.

I think we're not checking the `FLAGS_quic_reloadable_flag_quic_testonly_default_false` flag then afterwards as we set that true in the second block. but we check if 

`EXPECT_TRUE(GetQuicRestartFlag(quic_testonly_default_true));` is true which we didn't change in this process

Reply:
Oh, I see! The EXPECT on line 500 should be changed. Thanks, nice catch!

Done.